### PR TITLE
[PoC] inline svg files

### PR DESCRIPTION
**Pull request by rullzer**
Saves loading a lot of resources. And since the css file is cached
anyway this can save precious seconds.

Todo:
- [ ] CI job to check
- [ ] Look for more files to use
- [ ] Some places we use background-url but in some places we still use img stags. The img tags can't be repalced by this method.

Files so far:
- icons.scss

@ChristophWurst as discussed.

### CODE REVIEW

File: core/css/iconsinline.scss [hunk 1]
```diff
@@ -0,0 +1,473 @@
...
+ * @copyright Copyright (c) 2016, John Molakvoæ <skjnldsv@protonmail.com>
+ * @copyright Copyright (c) 2016, Joas Schilling <coding@schilljs.com>
+ * @copyright Copyright (c) 2016, Lukas Reschke <lukas@statuscode.ch>
+ * @copyright Copyright (c) 2016, Roeland Jago Douma <roeland@famdouma.nl>
+ * @copyright Copyright (c) 2016, Vincent Chan <plus.vincchan@gmail.com>
+ * @copyright Copyright (c) 2015, Thomas Müller <thomas.mueller@tmit.eu>
+ * @copyright Copyright (c) 2015, Hendrik Leppelsack <hendrik@leppelsack.de>
+ * @copyright Copyright (c) 2015, Jan-Christoph Borchardt <hey@jancborchardt.net>
+ *
+ * @license GNU AGPL version 3 or any later version
```
Comment (MorrisJobke) [inline core/css/iconsinline.scss]: Could we add a message here saying: "This file is automatically generated"?

File: build/inlineicons.php [hunk 2]
```diff
@@ -0,0 +1,32 @@
+<?php
```
Comment (MorrisJobke) [inline build/inlineicons.php]: Also execute this on CI and throw an error if there is a change. 😉

**Review by MorrisJobke — CHANGES_REQUESTED** [review]
Some refinements, but looks good code wise :)

**Comment by MorrisJobke** [comment]
This alone removes 13 of 129 requests of the files app (10%) 🙌

File: build/inlineicons.php (hunk 2 above)
Comment (rullzer) [inline build/inlineicons.php]: Yeah was thinking about that as well of course

**Comment by MorrisJobke** [comment]
May be an alternative to #3790

**Comment by jancborchardt** [comment]
We would need a way that the »build process« for this is automated. Also, the icons.scss file has other things than just .svg links. :)

**Comment by rullzer** [comment]
Lets kill this. We can only do this properly if we would have a compile step or something for it (like scss or javascript). For now not worth it I think...

**Comment by jancborchardt** [comment]
Ref https://github.com/nextcloud/server/issues/1066 if we want to work on anything like that again. :)

**Pull request closed without merging**